am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java

```diff
@@ -0,0 +1,143 @@
+
+        assertThat(result).containsExactly(ResourceDefinition.builder()
+            .resourceName(RESOURCE_NAME)
+            .resourceType(RESOURCE_TYPE)
+            .serviceName(SERVICE_NAME)
+            .build());
+    }
+
+    @Test
+    void shouldRetrieveResourceDefinitionWhenRecordExistsWithMultiplePermissions() {
+        Set<Permission> permissions = ImmutableSet.of(Permission.READ, Permission.CREATE);
+
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(ROOT_ATTRIBUTE, permissions));
+
+        List<ResourceDefinition> result =
+            service.getResourceDefinitionsWithCreatePermission(Collections.singleton(ROLE_NAME));
+
+        assertThat(result).containsExactly(ResourceDefinition.builder()
+            .resourceName(RESOURCE_NAME)
+            .resourceType(RESOURCE_TYPE)
+            .serviceName(SERVICE_NAME)
+            .build());
+    }
+
+    @Test
+    void shouldRetrieveResourceDefinitionWhenMultipleRecordsExistWithDifferentAttributes() {
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(
+            JsonPointer.valueOf("/adult"), CREATE_PERMISSION));
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(
+            JsonPointer.valueOf("/child"), CREATE_PERMISSION));
+
+        List<ResourceDefinition> result =
+            service.getResourceDefinitionsWithCreatePermission(Collections.singleton(ROLE_NAME));
+
+        assertThat(result).containsExactly(ResourceDefinition.builder()
+            .resourceName(RESOURCE_NAME)
+            .resourceType(RESOURCE_TYPE)
+            .serviceName(SERVICE_NAME)
+            .build());
+    }
+
+    @Test
+    void shouldReturnEmptyListWhenRecordExistsButNoCreatePermission() {
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(ROOT_ATTRIBUTE, READ_PERMISSION));
+
+        List<ResourceDefinition> result =
+            service.getResourceDefinitionsWithCreatePermission(Collections.singleton(ROLE_NAME));
+
+        assertThat(result).isEmpty();
+    }
+
+    @Test
+    void shouldReturnEmptyListWhenNoRecords() {
+        List<ResourceDefinition> result =
+            service.getResourceDefinitionsWithCreatePermission(Collections.singleton(ROLE_NAME));
+
+        assertThat(result).isEmpty();
+    }
+
+    @Test
+    void shouldRetrieveMultipleResourceDefinitionsWhenMultipleRecordExistsForTheSameRole() {
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(CREATE_PERMISSION, resource, ROLE_NAME));
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(CREATE_PERMISSION, resource2, ROLE_NAME));
+
+        List<ResourceDefinition> result =
+            service.getResourceDefinitionsWithCreatePermission(Collections.singleton(ROLE_NAME));
+
+        assertThat(result).containsExactly(resource2, resource);
+    }
+
+    @Test
+    void shouldRetrieveMultipleResourceDefinitionsWhenDefaultPermissionsExistForDifferentRoles() {
```

Comment:
Please simulate scenario where resource A has permission granted to same attribute for two different roles.

```
importerService.grantDefaultPermission(createDefaultPermissionGrant(CREATE_PERMISSION, resource, ROLE_NAME));
importerService.grantDefaultPermission(createDefaultPermissionGrant(CREATE_PERMISSION, resource, OTHER_ROLE_NAME));
```

I suspect that currently two duplicate items will be returned.

Reply:
Added test case. I believe I had already fixed this before I saw this comment.